contracts/epochs/FlowDKG.cdc

```diff
@@ -100,6 +90,230 @@ access(all) contract FlowDKG {
         }
     }
 
+    // Checks whether the ResultSubmission constructor arguments constitute a valid empty submission.
+    // A valid empty submission has all fields nil, and is used when the submittor locally failed the DKG.
+    // All non-empty submissions must have all non-nil fields.
+    access(all) view fun isValidNilSubmission(groupPubKey: String?, pubKeys: [String]?, idMapping: {String:Int}?): Bool {
+        // If any fields are nil, then this represents a empty submission and all fields must be nil
+        if groupPubKey == nil && pubKeys == nil && idMapping == nil {
+            return true
+        }
+        // Otherwise, all fields must be non-nil
+        return groupPubKey != nil && pubKeys != nil && idMapping != nil 
+    }
+
+    // Checks the group public key (part of a ResultSubmission) for validity.
+    // A valid public key in this context is either: (1) a hex-encoded 96 bytes, or (2) nil.
+    access(all) view fun isValidGroupKey(_ groupKey: String?): Bool {
+         if groupKey == nil {
+            // This is a nil/empty submission (see isValidNilSubmission)
+            return true
+        }
+        return groupKey!.length == FlowDKG.submissionKeyLength
+    }
+
+    // Checks a list of participant public keys (part of a ResultSubmission) for validity.
+    // A valid public key in this context is either: (1) a hex-encoded 96 bytes, or (2) nil.
+    access(all) view fun isValidPubKeys(_ pubKeys: [String]?): Bool {
+        if pubKeys == nil {
+            // This is a nil/empty submission (see isValidNilSubmission)
+            return true
+        }
+        for key in pubKeys! {
+            // keys must be exactly 96 bytes
+             if key.length != FlowDKG.submissionKeyLength {
+                 return false
+             }
+        }
+        return true
+    }
+
+    // Checks that an id mapping (part of ResultSubmission) contains one entry per public key.
+    access(all) view fun isValidIDMapping(pubKeys: [String]?, idMapping: {String: Int}?): Bool {
+        if pubKeys == nil {
```

Comment:
Should this be `pubKeys == nil && idMapping == nil`?

Reply:
We could check both fields, but we're relying on Invariant (1) enforced by the constructor precondition: either all fields are nil, or none are. (see comment on subsequent line)